packages/eslint-plugin/tests/rules/array-type.test.ts

```diff
@@ -2156,3 +2157,19 @@ type BrokenArray = {
     );
   });
 });
+
+describe('schema validation', () => {
+  // https://github.com/typescript-eslint/typescript-eslint/issues/6852
+  test("array-type does not accept 'simple-array' option", () => {
+    if (areOptionsValid(rule, [{ default: 'simple-array' }])) {
+      throw new Error(`Options succeeded validation for bad options`);
+    }
+  });
+
+  // https://github.com/typescript-eslint/typescript-eslint/issues/6892
+  test('array-type does not accept non object option', () => {
+    if (areOptionsValid(rule, ['array'])) {
+      throw new Error(`Options succeeded validation for bad options`);
+    }
+  });
+});
```

Comment:
I wonder if we should do a rule-tester style API here to remove the boilerplate.
For example something like this:

```ts
ruleOptionsTester(rule, {
  valid: [ ... ],
  invalid: [
    [{ default: 'simple-array' }],
    ['array'],
  ],
});
```

Reply:
Yeah, I think something like this would be neat if we end up doing a lot of these kinds of tests. For now I think it's simple enough that we can leave this to be done in future if necessary?

Reply:
Note that there is
https://github.com/eslint/rfcs/pull/103 and the implementing PR https://github.com/eslint/eslint/pull/16823

This would allow us to include these tests in the rule like:
```
tester.run('rule', rule, {
  valid: [ ... ],
  invalid: [ ... ],
  fatal: [
    {
      options: ['simple-array'],
      error: {
        name: 'SchemaValidationError',
        message: 'Whatever the json schema error is',
      },
    },
    {
      options: ['array'],
      error: {
        name: 'SchemaValidationError',
        message: 'Whatever the json schema error is',
      },
    },
  ],
}
```